float64 x_offset
float64 y_offset
float64 theta_offset
string state